### PR TITLE
Adds k8s-infra-staging-e2e-test-images@kubernetes.io group

### PR DESCRIPTION
**Pull request by bclau**
The group is meant to own pushing rights to the ``gcr.io/kubernetes-staging-e2e-test-images`` registry, which will contain the images to be promoted to the ``gcr.io/kubernetes-e2e-test-images`` registry.

### CODE REVIEW

File: k8s.gcr.io/k8s-staging-e2e-test-images/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# google group for gcr.io/kubernetes-staging-e2e-test-images is k8s-infra-staging-e2e-test-images@kubernetes.io
+registries:
+- name: gcr.io/kubernetes-staging-e2e-test-images
```
Comment (nikhita) [inline k8s.gcr.io/k8s-staging-e2e-test-images/manifest.yaml]: This would need to be `k8s-staging-e2e-test-images` to be consistent with other names, and also because the maximum number of characters in the name can only be 30.

Comment (bclau) [reply k8s.gcr.io/k8s-staging-e2e-test-images/manifest.yaml]: Done

File: infra/gcp/ensure-staging-storage.sh [hunk 2]
```diff
@@ -52,6 +52,7 @@ STAGING_PROJECTS=(
     coredns
     csi
     descheduler
+    e2e-test-images
```
Comment (thockin) [inline infra/gcp/ensure-staging-storage.sh]: We get one shot at naming.  Does "images" add enough value here, or is "e2e-test" or "test-e2e" or "testing/e2e" or even just "e2e" sufficient?

**Comment by Katharine** [comment]
@thockin https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/README.md#prow-config-template

**Comment by bclau** [comment]
I've sent the other 2 required PRs:

https://github.com/kubernetes/kubernetes/pull/84058
https://github.com/kubernetes/test-infra/pull/14833

**Comment by smourapina** [comment]
/milestone v1.17

Adding to milestone 1.17, as suggested in https://github.com/kubernetes/kubernetes/pull/81170#issuecomment-547433086
